x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -1,27 +1,52 @@
 lexer grammar EsqlBaseLexer;
 
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+@header {
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+}
+
+options {
+  superClass=LexerConfig;
+  caseInsensitive=true;
+}
+
 DISSECT : 'dissect'           -> pushMode(EXPRESSION_MODE);
 DROP : 'drop'                 -> pushMode(PROJECT_MODE);
 ENRICH : 'enrich'             -> pushMode(ENRICH_MODE);
 EVAL : 'eval'                 -> pushMode(EXPRESSION_MODE);
 EXPLAIN : 'explain'           -> pushMode(EXPLAIN_MODE);
 FROM : 'from'                 -> pushMode(FROM_MODE);
 GROK : 'grok'                 -> pushMode(EXPRESSION_MODE);
-INLINESTATS : 'inlinestats'   -> pushMode(EXPRESSION_MODE);
 KEEP : 'keep'                 -> pushMode(PROJECT_MODE);
 LIMIT : 'limit'               -> pushMode(EXPRESSION_MODE);
-LOOKUP : 'lookup'             -> pushMode(LOOKUP_MODE);
 META : 'meta'                 -> pushMode(META_MODE);
-METRICS : 'metrics'           -> pushMode(METRICS_MODE);
 MV_EXPAND : 'mv_expand'       -> pushMode(MVEXPAND_MODE);
 RENAME : 'rename'             -> pushMode(RENAME_MODE);
 ROW : 'row'                   -> pushMode(EXPRESSION_MODE);
 SHOW : 'show'                 -> pushMode(SHOW_MODE);
 SORT : 'sort'                 -> pushMode(EXPRESSION_MODE);
 STATS : 'stats'               -> pushMode(EXPRESSION_MODE);
 WHERE : 'where'               -> pushMode(EXPRESSION_MODE);
-MATCH : 'match'               -> pushMode(EXPRESSION_MODE);
-UNKNOWN_CMD : ~[ \r\n\t[\]/]+ -> pushMode(EXPRESSION_MODE);
+//
+// in development
+//
+INLINESTATS : {devVersion()}? 'inlinestats'   -> pushMode(EXPRESSION_MODE);
+LOOKUP :      {devVersion()}? 'lookup'        -> pushMode(LOOKUP_MODE);
+MATCH :       {devVersion()}? 'match'         -> pushMode(EXPRESSION_MODE);
+METRICS :     {devVersion()}? 'metrics'       -> pushMode(METRICS_MODE);
```

Comment:
I've moved the in-development commands separately and added a predicate for each of them so that they get picked up *ONLY* if devVersion returns true. If false, then the lexing fails - is as if the token hasn't been declared.
Or rather it has but it's an empty string (it gets assigned an token number however that one is never used).

The error message in the false case is `mismatched input 'inlinestats' expecting {'dissect', 'drop', 'enrich', 'eval', 'grok', 'keep', 'limit', 'mv_expand', 'rename', 'sort', 'stats', 'where', INLINESTATS, LOOKUP, MATCH}`

Notice the tokens specified at the end - I couldn't find a way to skip or move to a hidden channel these token from within the declaration itself; a workaround is to programatically remove the tokens so they don't appear in the error message.
However this is language specific - I'll follow up with a separate PR once we agree on this approach.

Reply:
We could rewrite the thing manually I guess.

Reply:
That `INLINESTATS` at the end of the error message IS confusing indeed.

Reply:
This has been addressed in the latest commit